code/backend/batch/utilities/integrated_vectorization/AzureSearchIndexer.py

```diff
@@ -42,9 +42,16 @@ def create_or_update_indexer(self, indexer_name: str, skillset_name: str):
         )
         return indexer_result
 
-    # To be updated for 'Reprocess All'
-    def run_indexer(self, indexer_name: str):
-        self.indexer_client.run_indexer(indexer_name)
-        logger.info(
-            f" {indexer_name} is created and running. If queries return no results, please wait a bit and try again."
-        )
+    def reprocess_all(self, indexer_name: str):
+        reprocess_response = {"status": "success"}
+        if indexer_name in [name for name in self.indexer_client.get_indexer_names()]:
+            self.indexer_client.reset_indexer(indexer_name)
+            run_response = self.indexer_client.run_indexer(indexer_name)
+            logger.info(
+                f" {indexer_name} is created and running. If queries return no results, please wait a bit and try again."
+            )
+            return reprocess_response if run_response is None else {"status": "error"}
+        else:
+            logger.error(f"Indexer {indexer_name} not found.")
+            reprocess_response = {"status": "error"}
```

Comment:
Could this not create the indexer if it doesn't exist instead of erroring?

Reply:
Yeah it can create, but I was thinking if there is no documents why do we create a new indexer? We are creating indexer when a document is ingested. This check was mostly to prevent if someone directly lands on the Admin page & clicks Reprocess button

Reply:
Could it be that there are documents, but the indexer has been deleted for some reason?

Reply:
Nowhere we are deleting the indexer, except if the user goes to the portal and deletes the indexer.

Reply:
Yea that's the scenario I'm thinking of

Reply:
I will add the creation index here then. Because otherwise they will have to ingest a document to get the indexer created